src/PortalGunHooks.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+import * as React from 'react';
+
+class Store {
+  static id = 100000;
+
+  listeners = [];
+  idsToNodes = new Map();
+
+  addNode(node, id) {
```

Comment:
I'd swap argument order; it's normally `key, value` for arguments (e.g. see the following 'set' call).